torch/quantization/observer.py

```diff
@@ -134,7 +134,7 @@ def _calculate_per_channel_qparams(self, min_vals, max_vals):
                 "must run observer before calling calculate_qparams.\
                                     Returning default scale and zero point "
             )
-            return torch.tensor([1.0]), torch.tensor([0])
+            return torch.tensor([1.0]), torch.tensor([0]), self.ch_axis
```

Comment:
Not sure I follow the reasoning here: The axis is passed as an attribute to observer and is used when statistics are collected. CalcQparams should be independent of the axis argument, it should know if per channel or per tensor calculation is desired, but operations are independent of the axis argument.

Reply:
we don't need axis for calculate_qparams, but we need the axis argument for per channel quantize op. so it's an output value, not an input argument to calculate_qparams.

Reply:
we'll have a separate `get_qparam` function that returns these things, I'll do this in a separate PR